nestkernel/event_delivery_manager.cpp

```diff
@@ -426,10 +426,6 @@ EventDeliveryManager::gather_spike_data_( const thread tid,
     const bool deliver_completed = deliver_events_( tid, recv_buffer );
     gather_completed_checker_[ tid ].logical_and( deliver_completed );
 
-// Exit gather loop if all local threads and remote processes are
-// done.
-#pragma omp barrier
-
```

Comment:
With barrier, we measure for deliver_spike_data the runtime of the slowest thread; without barrier, we measure most likely the average runtime over all threads. I think, both variants are ok.

Reply:
Code provisions made for benchmarking should not reduce speed under production conditions, so I think it is sensible to remove this barrier.

Reply:
Would this fall into the category of performance instrumentation that was discussed some time ago? We could make this barrier dependent on a cmake flag, but it might also confuse more than it helps. In any case, for sure this important difference in measurement must also be expressed in the documentation of the kernel timers!
@stinebuu, could you check if this difference is clear enough in the docs?

Would this be a problem for comparing benchmarks between NEST versions with/without this barrier?